CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/dispatcher/DispatchersProviderImpl.kt

```diff
@@ -0,0 +1,14 @@
+package co.nimblehq.coroutine.dispatcher
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainCoroutineDispatcher
+
+class DispatchersProviderImpl : DispatchersProvider {
+
+    override fun getIO(): CoroutineDispatcher = Dispatchers.IO
+
+    override fun getMain(): MainCoroutineDispatcher = Dispatchers.Main
```

Comment:
I think we can confront with the Interface return type to return explicitly `CoroutineDispatcher` as defined from the interface; upcasting in this case doesn't hurt imo, instead of returning a specific type `MainCoroutineDispatcher`?